File: python/shippable_product.py
from product import Product
from shippable import Shippable

class ShippableProduct(Product, Shippable):
    def __init__(self, name, price, quantity, weight):
        super().__init__(name, price, quantity)
        self.weight = weight
    
    def get_weight(self):
        return self.weight
    
    def requires_shipping(self):
        return True

File: cpp/ShippableProduct.cpp
// cpp/ShippableProduct.cpp
#include "ShippableProduct.h"

ShippableProduct::ShippableProduct(std::string name, double price, int quantity, double weight)
    : Product(name, price, quantity), weight(weight) {}

double ShippableProduct::getWeight() const {
    return weight;
}

bool ShippableProduct::requiresShipping() const {
    return true;
}
